buffer/src/main/java/io/netty/buffer/UnsafeByteBufUtil.java

```diff
@@ -431,7 +431,7 @@ private static void batchSetZero(byte[] data, int index, int length) {
             PlatformDependent.putLong(data, index, ZERO);
             index += 8;
         }
-        final int remaining = length % 8;
+        final int remaining = length & 0x07;
```

Comment:
Just above both of these, we have a `length / 8` as well. Does it make sense to change those to shifts?

Reply:
I didn't checked but in few parts of Netty (e.g. zeroing and indexOf..I remember there are division and modulus by power of 2; can you check if we need the same treatment?

Reply:
You're right, there is `addr % 8`, but `addr` can be negative as I undertandd it, which makes substitution not correct.